internal/fs/ea_windows.go

```diff
@@ -0,0 +1,284 @@
+//go:build windows
+// +build windows
+
+package fs
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// The code below was adapted from https://github.com/microsoft/go-winio under MIT license.
+
+// The MIT License (MIT)
+
+// Copyright (c) 2015 Microsoft
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// The code below was copied over from https://github.com/microsoft/go-winio/blob/main/ea.go under MIT license.
```

Comment:
The important code that is being used in our _ea_windows.go_ was taken from a PR of go-winio - https://github.com/microsoft/go-winio/pull/229

This PR was never merged into go-winio, because they no longer needed those functions. They are using `PAXHeaders` for some tar related functionality and only use from encoding/decoding functions for EA in their _ea.go_.

Also, the code that we need like `GetFileEA` uses a private struct `ioStatusBlock` and private function `ntStatus` which are defined in go-winio. Hence, even if we import go-winio we would have to redefine that code in our _ea.go_.
The entire test case is from that unmerged PR as well.

Still, by importing go-winio, we will be able to use `winio.EncodeExtendedAttributes` and `winio.DecodeExtendedAttributes`.

_ea_windows.go_ would reduce from **285 lines to 140 lines**.


In _sd_windows.go_ we would be able to directly use `winio.EnableProcessPrivileges` and all the privilege related functions would be handled in go-winio.

_sd_windows_ would reduce from **440 lines to 189 lines**.  

Also, I see that by adding the go-winio module and making all the necessary changes, the windows binary size increases by 10 KB only.

So, it might make sense to use go-winio. Would you like me to make the changes?

Reply:
How about the following: let's merge this PR with the duplicated code and import go-winio in a follow-up PR. My plan is to release restic 0.17.0 in a few weeks and after that is done, I'll bump the go requirement to 1.21. Then we could directly start with using 0.6.2